src/rtcModule/webrtcAdapter.h

```diff
@@ -250,190 +255,161 @@ class myPeerConnection: public
-            mAudio->track()->set_enabled(audio);
-        if (mVideo && mVideo->track()->enabled() != video)
-            mVideo->track()->set_enabled(video);
+    bool is_screencast() const override { return false; }
+    absl::optional<bool> needs_denoising() const override {
+        return absl::nullopt;
     }
-    LocalStreamHandle(const std::shared_ptr<LocalAudioTrackHandle>& aAudio,
-        const std::shared_ptr<LocalVideoTrackHandle>& aVideo, const char* name="localStream")
-    :mAudio(aAudio), mVideo(aVideo), mStream(gWebrtcContext->CreateLocalMediaStream(name))
+
+    bool GetStats(webrtc::VideoTrackSourceInterface::Stats* stats) override { return false; }
+
+    void AddOrUpdateSink(rtc::VideoSinkInterface<webrtc::VideoFrame>* sink,
+                         const rtc::VideoSinkWants& wants) override;
+    void RemoveSink(rtc::VideoSinkInterface<webrtc::VideoFrame>* sink) override;
+    virtual bool remote() const override;
+    virtual void OnFrame(const webrtc::VideoFrame& frame) override;
+    virtual void AddRef() const override
     {
-        if (!mStream.get())
-            throw std::runtime_error("MyStream: Error creating stream object");
-        bool ok = true;
-        if (aAudio)
-            ok &= mStream->AddTrack(*aAudio);
-        if (aVideo)
-            ok &= mStream->AddTrack(*aVideo);
-        if (!ok)
-            throw std::runtime_error("Error adding track to media stream");
+        mRefCount.IncRef();
     }
-    ~LocalStreamHandle()
-    { //make sure the stream is released before the tracks
-        mStream = nullptr;
+
+    virtual rtc::RefCountReleaseStatus Release() const override
+    {
+        const auto status = mRefCount.DecRef();
+        if (status == rtc::RefCountReleaseStatus::kDroppedLastRef) {
+          delete this;
+        }
+
+        return status;
     }
-    webrtc::AudioTrackInterface* audio() { return mAudio?*mAudio:(webrtc::AudioTrackInterface*)nullptr; }
-    webrtc::VideoTrackInterface* video() { return mVideo?*mVideo:(webrtc::VideoTrackInterface*)nullptr; }
-    tspMediaStream& stream() { return mStream; }
-    operator webrtc::MediaStreamInterface*() { return mStream; }
-    operator const webrtc::MediaStreamInterface*() const { return mStream; }
+
+    static std::set<std::pair<std::string, std::string>> getVideoDevices();
+    virtual void openDevice(const std::string &videoDevice) override;
+    virtual void releaseDevice() override;
+    virtual rtc::scoped_refptr<webrtc::VideoTrackSourceInterface> getVideoTrackSource() override;
+
+protected:
+    rtc::ThreadChecker mWorkerThreadChecker;
+    rtc::VideoBroadcaster mBroadcaster;
+    webrtc::MediaSourceInterface::SourceState mState;
+    bool mRemote;
+    rtc::scoped_refptr<webrtc::VideoCaptureModule> mCameraCapturer;
+    webrtc::VideoCaptureCapability mCapabilities;
+    mutable webrtc::webrtc_impl::RefCounter mRefCount{0};
 };
 
-class DeviceManager
+class CapturerTrackSource : public VideoManager
```

Comment:
Probably this class can be removed in pro of a design where the `VideoManager` itself inherits from `VideoTrackSourceInterface`, which would result in a cleaner and simpler design.
Furthermore, it should help, as well, to avoid the issue with the destruction of the implementation of such interface, which is an object shared between Objective C and C++ / WebRTC.